kernel/cond.c

```diff
@@ -64,6 +65,14 @@ int myst_cond_timedwait(
         /* Add the self thread to the end of the wait queue */
         myst_thread_queue_push_back(&c->queue, self);
 
+        // assert(self->signal.cond_wait == NULL);
```

Comment:
Why comment out this assert?

Reply:
It was failing before the final fix. I am not sure if we should enable it now.